.github/workflows/experimental.yaml

```diff
@@ -0,0 +1,119 @@
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: go install ./cmd/affected
+        working-directory: .github/workflows/samples-tools
+      - name: Get diffs
+        run: git --no-pager diff --name-only HEAD origin/main | tee diffs.txt
+      - name: Find Node.js affected packages
+        id: nodejs
+        run: echo "packages=$(affected -config .github/workflows/config/nodejs.jsonc -diffs=diffs.txt)" | tee -a $GITHUB_OUTPUT
+
+  nodejs-lint:
+    name: Node.js lint
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: make lint
+      - run: ./.github/workflows/utils/region-tags-tests.sh
+
+  nodejs-test:
+    name: Node.js 20 test
+    needs: affected
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # needed for google-github-actions/auth
+    strategy:
+      fail-fast: false
+      matrix:
+        package: ${{ fromJson(github.event_name == 'pull_request' && needs.affected.outputs.nodejs || '[]') }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - uses: google-github-actions/auth@v2
+        with:
+          project_id: long-door-651
+          workload_identity_provider: projects/1046198160504/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider
+          service_account: kokoro-system-test@long-door-651.iam.gserviceaccount.com
+          access_token_lifetime: 600s # 10 minutes
+      - run: make test dir=${{ matrix.package }}
+        env:
+          GOOGLE_SAMPLES_PROJECT: long-door-651
+
+  # nodejs-nightly:
+  #   name: Node.js 20
+  #   runs-on: ubuntu-latest
+  #   permissions:
+  #     id-token: write # needed for google-github-actions/auth
+  #   strategy:
+  #     matrix:
+  #       run: ${{ fromJson(github.event_name == 'schedule' && '["nightly tests"]' || '[]') }}
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: actions/setup-node@v4
+  #       with:
+  #         node-version: 20
+  #     - uses: actions/setup-go@v5
+  #       with:
+  #         go-version: ${{ env.GO_VERSION }}
+  #     - uses: google-github-actions/auth@v2
+  #       with:
+  #         project_id: long-door-651
+  #         workload_identity_provider: projects/1046198160504/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider
+  #         service_account: kokoro-system-test@long-door-651.iam.gserviceaccount.com
+  #         access_token_lifetime: 600s # 10 minutes
+  #     - run: go install ./cmd/run-all
+  #       working-directory: .github/workflows/samples-tools
+  #     - run: run-all -config=.github/workflows/config/nodejs.jsonc make clean-test dir=%s
```

Comment:
thought: there are a lot of arguments to `run-all`. Can we possibly remove some that are duplicative? Or maybe assume some defaults--it seems to me that the config is likely going to be in a default location.

Reply:
We might have different configs for different kinds of runs. For example, end-to-end tests will most likely have their own config since they do more custom setup. Also the tool should work for all other repositories, so they will likely have their own config files as well. And if we end up going to monorepo, each language would have their own config.   I could even see things like AppEngine samples having their own config in Python since they're the only ones running on 2.7.

I also don't like the verbosity, but overall I like that it's explicit and we can immediately know which files are related without having to dig into the tooling code (which will live in its own repo at some point).

Reply:
I think we could use a reusable workflow, where we customize the commands to run.